Use device tree to generate CONFIG_UART_CONSOLE_ON_DEV_NAME defines

galak opened the pull request.

galak committed:
board: cc2650_sensortag: Add zephyr,console reference to device tree

Since we support uart0 on the cc2650 sensortag we should have a
reference for zephyr,console in chosen for it.  This will be useful as
we start to generate Kconfig defines based on zephyr,console.

Signed-off-by: Kumar Gala <kumar.gala@linaro.org>

agross-linaro commented on scripts/extract_dts_includes.py:
```diff
@@ -404,6 +404,20 @@ def extract_single(node_address, yaml, prop, key, prefix, defs, def_label):
+    defs[node_address].update(prop_def)
+  else:
+    defs[node_address] = prop_def
+
```
I wonder if the extract_property shouldnt be more generic and use the type to decide what to do about the value.  As it is right now, the extract_single is being called within the extract_property.  If extract single handled the stringification, we'd be ok to just use that.  The only other issue to deal with is the label.

galak replied:
hmm, need to look at that.  What exactly is extract_property trying to do.  The stringification seems to happen ok in extract_single, but what's going on with the label from the yaml I don't quite understand.

agross-linaro replied:
it basically steers the handling of the property.  The question is whether or not we should really be keying off the yaml to figure out the parsing.

galak replied:
What if we merge this as is, and figure out how to refactor extract_property later?

agross-linaro replied:
i have no problem with that.

agross-oss reviewed (commented):
All of it looks fine except for my questions concerning the addition of the extract string property function.

galak committed:
devicetree: Generate CONFIG_UART_CONSOLE_ON_DEV_NAME from dt

We can use the chosen property "zephyr,console" to determine what uart
should be used as the console and find its name to generate a define for
CONFIG_UART_CONSOLE_ON_DEV_NAME.

Signed-off-by: Kumar Gala <kumar.gala@linaro.org>

SavinayDharmappa commented:
Is this PR is merged ? I don't see change after pull.

galak commented:
@SavinayDharmappa merged now
